### PR TITLE
[FEAT] RLHF on response messages

### LINKED ISSUE #595: [FEAT]:  RHLF like feature.
### What would you like to see?

An option for Vetting the queries-answers by different users and send it to the workspace embedding. I have this immense resource of question answers that can be used further to train the LLM - although I don't want to go fine-tuning way.

An option for the user to give thumbs-up or down; visible to the manager/ administrator in Workspace Chat pane - and the Administrator can send (after some editing) this to workspace embeddings so that some kind of RHLF happens. But these should be kept separate from the documents uploaded - for better folder management. Thanks

### PR DESCRIPTION
**Pull request by shatfield4**
### Pull Request Type

- [x] ✨ feat
- [ ] 🐛 fix
- [ ] ♻️ refactor
- [ ] 💄 style
- [ ] 🔨 chore
- [ ] 📝 docs

### Relevant Issues

resolves #595 

### What is in this change?

Describe the changes in this PR that are impactful to the repo.

- Implement RLHF in chats where user can thumbs up or down messages
- Create column in workspace_chats table to keep track of feedback score

### Additional Information

Add any other context about the Pull Request here that was not captured above.

### Developer Validations

- [x] I ran `yarn lint` from the root of the repo & committed changes
- [x] Relevant documentation has been updated
- [x] I have tested my code functionality
- [x] Docker build succeeds locally

### CODE REVIEW

**Comment by shatfield4** [comment]
Currently works when refreshing page after sending chat or loading existing chat from historical messages.

WIP - Fix for after chat message is sent it needs to pass the chatId from last message to Actions component in order to update result.

File: frontend/src/components/WorkspaceChat/ChatContainer/ChatHistory/HistoricalMessage/index.jsx [hunk 1]
```diff
@@ -12,7 +12,16 @@ import createDOMPurify from "dompurify";
...
   (
-    { uuid = v4(), message, role, workspace, sources = [], error = false },
+    {
+      uuid = v4(),
+      message,
+      role,
+      workspace,
+      sources = [],
+      error = false,
+      feedbackScore,
```
Comment (timothycarambat) [inline frontend/src/components/WorkspaceChat/ChatContainer/ChatHistory/HistoricalMessage/index.jsx:22]: set default value, even if null

File: frontend/src/models/workspace.js [hunk 2]
```diff
@@ -60,6 +60,16 @@ const Workspace = {
       .catch(() => []);
     return history;
   },
+  updateChatFeedback: async function (chatId, feedback) {
+    const result = await fetch(`${API_BASE}/workspace/chat-feedback`, {
+      method: "POST",
+      headers: baseHeaders(),
+      body: JSON.stringify({ chatId, feedback }),
```
Comment (timothycarambat) [inline frontend/src/models/workspace.js:67]: add chatID as url param with workspace slug as well.

Then we can pre-check the workspace x user ownership and then update on backend.

File: server/endpoints/workspaces.js [hunk 3]
```diff
@@ -321,6 +321,26 @@ function workspaceEndpoints(app) {
...
 
+  app.post(
+    "/workspace/chat-feedback",
+    [validatedRequest],
+    async (request, response) => {
+      try {
+        const { chatId, feedback } = reqBody(request);
+        const result = await WorkspaceChats.updateFeedbackScore(
+          Number(chatId),
+          Number(feedback)
```
Comment (timothycarambat) [inline server/endpoints/workspaces.js:332]: Do the casting inside of the method here, would prevent accidental non-cast if we use it later.

**Pull request merged**